### PR TITLE
Add utility function to refresh shoot webhook netpols

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area networking
/kind bug
/priority normal

**What this PR does / why we need it**:
A utility function that reconciles the shoot webhook configurations has been added to the generic `ControlPlane` actuator package. It is recommended for all extensions that have shoot webhooks to call this function before starting the control loops to ensure that the webhook configurations are updated in case the ports change.

**Special notes for your reviewer**:
/cc @timebertt @stoyanr 

**Release note**:

```noteworthy developer
A utility function that reconciles the shoot webhook configurations has been added to the generic `ControlPlane` actuator package. It is recommended for all extensions that have shoot webhooks to call this function before starting the control loops to ensure that the webhook configurations are updated in case the ports change.
```

### CODE REVIEW

File: extensions/pkg/controller/controlplane/genericactuator/actuator.go [hunk 1]
```diff
@@ -525,3 +506,69 @@ func (a *actuator) Migrate(
...
+	}); err != nil {
+		return err
+	}
+
+	fns := make([]flow.TaskFn, 0, len(namespaceList.Items))
+
+	for _, namespace := range namespaceList.Items {
+		var (
+			networkPolicy = extensionswebhookshoot.GetNetworkPolicyMeta(namespace.Name, providerName)
+			namespace     = networkPolicy.Namespace
```
Comment (vpnachev) [inline extensions/pkg/controller/controlplane/genericactuator/actuator.go]: This variable is already set and it has the same value, so I think it is not necessary to redefine it here, wdyt?

Comment (rfranzke) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go]: Yeah, I wasn't entirely sure whether it's safe to use the iteration variable when the functions are added, so I declared these variables.
/invite @timuthy

Comment (rfranzke) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go]: I remember we had some issues with that approach, e.g., [here](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/containerruntime.go#L47)

Comment (vpnachev) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go]: In this case I would just rename it to not be the same as the loop variable.

Comment (vpnachev) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go]: I had to refresh, my bad, sorry.

**Review by vpnachev — DISMISSED** [review]
/lgtm

File: extensions/pkg/controller/controlplane/genericactuator/actuator.go [hunk 2]
```diff
@@ -525,3 +506,65 @@ func (a *actuator) Migrate(
...
+		return err
+	}
+
+	fns := make([]flow.TaskFn, 0, len(namespaceList.Items))
+
+	for _, namespace := range namespaceList.Items {
+		networkPolicy := extensionswebhookshoot.GetNetworkPolicyMeta(namespace.Name, providerName)
+
+		fns = append(fns, func(ctx context.Context) error {
+			if err := c.Get(ctx, kutil.Key(namespace.Name, networkPolicy.Name), &networkingv1.NetworkPolicy{}); err != nil {
```
Comment (timuthy) [inline extensions/pkg/controller/controlplane/genericactuator/actuator.go]: I still think the ref to the `namespace` loop var causes issues. See this example https://play.golang.org/p/_ehBa__iRC_e.

**Review by timuthy — CHANGES_REQUESTED** [review]

File: extensions/pkg/controller/controlplane/genericactuator/actuator.go (hunk 2 above)
Comment (rfranzke) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go]: Yep, thanks Tim, I've adapted the code and also implemented a unit test for the function. Please check.

**Review by timuthy — APPROVED** [review]
/lgtm

File: extensions/pkg/controller/controlplane/genericactuator/actuator.go [hunk 3]
```diff
@@ -525,3 +506,69 @@ func (a *actuator) Migrate(
...
+			networkPolicyName = networkPolicy.Name
+		)
+
+		fns = append(fns, func(ctx context.Context) error {
+			if err := c.Get(ctx, kutil.Key(namespaceName, networkPolicyName), &networkingv1.NetworkPolicy{}); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
```
Comment (stoyanr) [inline extensions/pkg/controller/controlplane/genericactuator/actuator.go:568]: This code skips the webhook reconciliation if the networkpolicy was not found. Why is this needed / when would this happen? Isn't it better to reconcile anyway, just in case?

**Review by stoyanr — COMMENTED** [review]
Just a small question from my side.

File: extensions/pkg/controller/controlplane/genericactuator/actuator.go (hunk 3 above)
Comment (rfranzke) [reply extensions/pkg/controller/controlplane/genericactuator/actuator.go:568]: For example, when a shoot is already in the process of being deleted and the webhook configuration has already been cleaned up (then we don't want to recreate it unnecessarily).

**Review by vpnachev — APPROVED** [review]
/lgtm
